SampleApps/SampleApps/SampleEnclave/my_exports.cpp

```diff
@@ -283,6 +332,253 @@ HRESULT VbsEnclave::Trusted::Implementation::RunTaskpoolExample(_In_ const std::
+
+        debug_print(L"Created secure cache configuration in VTL1");
+
+        auto keyBytes = veil::vtl1::userboundkey::create_user_bound_key(
+            helloKeyName,
+            secureConfig,
+            pinMessage,
+            windowId,
+            ENCLAVE_SEALING_IDENTITY_POLICY::ENCLAVE_IDENTITY_POLICY_SEAL_SAME_IMAGE,
+            ENCLAVE_RUNTIME_POLICY_ALLOW_FULL_DEBUG,
+            keyCredentialCacheOption);
+
+        debug_print(L"create_user_bound_key returned");
+
+        // Store the user-bound key bytes directly - do NOT try to create a symmetric key from them
+        securedEncryptionKeyBytes.assign(keyBytes.begin(), keyBytes.end());
+        
+        // Do NOT try to create a symmetric key here - user-bound keys must be loaded properly
+        // g_encryptionKey will be set in the load functions
+
+        return S_OK;
+    }
+    CATCH_RETURN();
+}
+
+//
+// Load user bound key and encrypt data
+//
+HRESULT VbsEnclave::Trusted::Implementation::MyEnclaveLoadUserBoundKeyAndEncryptData(
+    _In_ const std::wstring& helloKeyName,
+    _In_ const std::wstring& pinMessage,
+    _In_ const uintptr_t windowId,
+    _In_ const std::vector<std::uint8_t>& securedEncryptionKeyBytes,
+    _In_ const std::wstring& inputData,
+    _Out_ std::vector<std::uint8_t>& combinedOutputData)
+{
+    using namespace veil::vtl1::vtl0_functions;
+
+    try
+    {
+        debug_print(L"Start MyEnclaveLoadUserBoundKeyAndEncryptData");
+
+        // Only load the user-bound key if it's not already loaded
+        if (!IsUBKLoaded())
+        {
+            debug_print(L"UBK not loaded, loading user-bound key");
+
+            // VTL1 creates secure cache configuration - VTL0 input is ignored
+            auto secureConfig = CreateSecureKeyCredentialCacheConfig();
+
+            debug_print(L"Created secure cache configuration in VTL1");
+
+            // Load the user-bound key from secured encryption key bytes
+            // Note: securedEncryptionKeyBytes contains the user-bound key data, not raw key material
+            auto loadedKeyBytes = veil::vtl1::userboundkey::load_user_bound_key(
+                helloKeyName,
+                secureConfig,
+                pinMessage,
+                windowId,
+                securedEncryptionKeyBytes);
+
+            debug_print(L"load_user_bound_key returned, key size: %d", loadedKeyBytes.size());
+
+            // NOW we can create a symmetric key from the loaded raw key material
+            auto newEncryptionKey = veil::vtl1::crypto::create_symmetric_key(loadedKeyBytes);
+            SetEncryptionKey(std::move(newEncryptionKey));
+            debug_print(L"Created symmetric key from loaded user-bound key material");
+        }
+        else
+        {
+            debug_print(L"UBK already loaded, using cached key");
+        }
+
+        // Use the global key for encryption
+        debug_print(L"Encrypting input data");
+        auto keyHandle = GetEncryptionKeyHandle();
```

Comment:
>GetEncryptionKeyHandle [](http://example.com/codeflow?start=25&length=22)

We probably don't need IsUBKLoaded if we have this one.    "TryGetEncryptionKeyHandle" does both things.

(I'd remove the check on line 429)